sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/change_feed_state.py

```diff
@@ -367,12 +365,18 @@ def from_continuation(
         if continuation_data is None:
             raise ValueError(f"Invalid continuation: [Missing {ChangeFeedStateV2.continuation_property_name}]")
         continuation = FeedRangeCompositeContinuation.from_json(continuation_data)
-        return ChangeFeedStateV2(
+
+        mode = continuation_json.get(ChangeFeedStateV2.mode_property_name)
+        if mode is None:
+            raise ValueError(f"Invalid continuation: [Missing {ChangeFeedStateV2.mode_property_name}]")
```

Comment:
This would be breaking if someone presents a ChangeFeedStateV2 continuation from an earleir SDK version, correct? That is a breaking change - change feed is often long running - so deifnitely possible that continuation tokens form previous SDK versions are presented. If I am missing something and this would work somehow please add brief comment why it is safe and add a tests case where we capture the json of a continuation how it looked before this and then use that to validate old continuation tokens continue to work.

Reply:
It is safe as far as I know of. @xinlian12  Please correct me if I am wrong.

Basically, `from_continuation()` only gets called if the change feed state version was `V2` in continuation. Ever since `ChangeFeedStateV2` was added(by @xinlian12), it has been creating continuation tokens included the following 2 properties (https://github.com/Azure/azure-sdk-for-python/pull/36930/files#diff-2d46bcf89a9e926a031b2c470534f67c62d9da61175a8191b74c088ea8aa36aeR220)
- version_property_name = "v"
- mode_property_name = "mode"

If ChangeFeedState version was `v2`, continuation must contain `mode` property. If not, the continuation had been corrupted, we must raise an exception with missing property.

For older continuation tokens, the ChangeFeedState vesions shouldn't be v2, so they would never hit this point.

Also for the test cases, I already added the test with corrupted continuation token: https://github.com/Azure/azure-sdk-for-python/pull/38105/files#diff-8b3af0ea28342f506eb8698103b130f4c85beba28fbc17750506ba0359d30f48R435

Reply:
I also added the comment as you suggested

Reply:
yea, if the customer is using the old continuationToken(like just the lsn), it will go through a different code path. 

when the new continuation token format being added, it always contain a "mode" property and default to LatestVersion. In this PR, the only thing changed is the value can be in the mode property: either LatestVersion or AllVersionsAndDeletes

Reply:
Ahh I missed line 221 in the old file - where mode was set to a constant. Makes sense.